.github/workflows/test_osdk.yml

```diff
@@ -42,3 +46,38 @@ jobs:
           cd osdk
           RUSTUP_HOME=/root/.rustup cargo +stable build
           RUSTUP_HOME=/root/.rustup cargo test
+  osdk-test-tdx:
+    if: github.event_name == 'schedule'
+    runs-on: self-hosted
+    timeout-minutes: 30
+    env:
+      # Need to set up proxy since the self-hosted CI server is located in China,
+      # which has poor network connection to the official Rust crate repositories.
+      RUSTUP_DIST_SERVER: https://mirrors.ustc.edu.cn/rust-static
+      RUSTUP_UPDATE_ROOT: https://mirrors.ustc.edu.cn/rust-static/rustup
+    strategy:
+      matrix:
+        # asterinas/asterinas:0.6.2-tdx container is the developing container of asterinas,
+        # asterinas/osdk:0.6.2-tdx container is built with the intructions from Asterinas Book
+        container: ['asterinas/asterinas:0.6.2-tdx', 'asterinas/osdk:0.6.2-tdx']
+    container: 
+      image: ${{ matrix.container }}
+      options: --device=/dev/kvm --privileged
+    steps:
+      - run: echo "Running in ${{ matrix.container }}"
+      - uses: actions/checkout@v4
+      - name: Lint
+        id: lint
+        if: matrix.container == 'asterinas/asterinas:0.6.2-tdx'
+        run: make check_osdk
+      # Github's actions/checkout@v4 will result in a new user (not root) 
+      # and thus not using the Rust environment we set up in the container. 
+      # So the RUSTUP_HOME needs to be set here. 
+      # This only breaks when we invoke Cargo in the integration test of OSDK 
+      # since the OSDK toolchain is not nightly.
```

Comment:
We currently accept OSDK to be built and tested using the nightly toolchain, so you may remove the `+stable` here and modify the comments accordingly. Modify the non-OSDK test as well.